static/templyte/src/app/download-template/download-template.component.ts

```diff
@@ -64,6 +74,43 @@ export class DownloadTemplateComponent implements OnInit {
         this.authService.logout(null);
     }
 
+    storeRating(new_rating) {
+        const component = this;
+        console.log("new rating: "+new_rating);
+        let authorUID:string  =  component.user.uid;
+        component.templateRatingsInfoDatabaseRef.once('value').then(snapshot2 => {
+            const old_rating = snapshot2.child(authorUID).val(); //value of previous rating
+            console.log("old_rating: "+old_rating);
+            let hasRated:boolean = old_rating != null;
+            if (hasRated == true) {
```

Comment:
1. Be very careful about using '==' in JavaScript; '===' is usually preferable, unless you're trying to check equality of addresses, etc.
2. Is there a reason you're not using 'if(hasRated)'? If so, it should be in a comment, because any reasonable maintainer would likely make the change without worrying about potential consequences.
3. Why not just inline this?